torch/ao/quantization/fx/prepare.py

```diff
@@ -514,14 +493,8 @@ def maybe_insert_input_observer_for_arg_or_kwarg(
             (arg_as_output_target_dtype not in DO_NOT_OBS_DTYPE_LIST) and
             # if qconfig is reuse_input qconfig, we won't insert extra observer for input
             not is_reuse_input_qconfig_ or
-            # need to add input observer for dynamic quantization
-            # only add observer for first input for now, we may need to extend
-            # qconfig_dict and backend_config_dict to support more general configurations
-            # of dynamic quantization, e.g. dynamically quantizing second input, third
-            # input etc.
-            (arg_as_input_target_compute_dtype in [torch.quint8, torch.int8, torch.float16]) and arg is node.args[0]
+            qconfig.activation().replacement_quant_is_dynamic and arg is node.args[0]
```

Comment:
is `arg is node.args[0]` needed?

Reply:
oh I think it's fine to keep this, since the qconfig is assuming this right now

Reply:
maybe we can return `replacement_quant_is_dynamic` from `get_qconfig_dtypes` and store this information together with other dtype information, in L358, so that we don't need to query qconfig again here.